src/tests/functional/shared_test_classes/src/single_op/generate_proposals.cpp

```diff
@@ -91,5 +93,63 @@ void GenerateProposalsLayerTest::SetUp() {
         params,
         "GenerateProposals");
 }
+
+void GenerateProposalsLayerTest::compare(const std::vector<ov::Tensor>& expected,
+                                         const std::vector<ov::Tensor>& actual) {
+    if (targetDevice != ov::test::utils::DEVICE_GPU) {
+        SubgraphBaseTest::compare(expected, actual);
+        return;
+    }
+
+    const auto outputsNum = expected.size();
+    ASSERT_EQ(outputsNum, 3);
+    ASSERT_EQ(outputsNum, actual.size());
+    ASSERT_EQ(outputsNum, function->get_results().size());
+
+    // actual outputs 0 (rois) and 1 (roi_scores) may be padded with zeros
+    for (size_t i = 0; i < 2; ++i) {
+        const auto expectedNumRois = expected[i].get_shape()[0];
+        const auto actualNumRois = actual[i].get_shape()[0];
+        ASSERT_LE(expectedNumRois, actualNumRois);
+
+        const auto actualBuffer = static_cast<uint8_t*>(actual[i].data());
+        const auto expectedBuffer = static_cast<uint8_t*>(expected[i].data());
+        const auto outputSize = i == 0 ? 4 : 1;
+
+        if (outType == ov::element::f32) {
+            LayerTestsUtils::LayerTestsCommon::Compare(reinterpret_cast<const float*>(expectedBuffer),
```

Comment:
please, don't use legacy tests infrastructure, it will be deleted soon. Try to use `ov::test::utils::compare`

Reply:
Part of outputs can be zero. ov::test::utils::compare without size parameter